PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamPairCleaner.cxx

```diff
@@ -202,6 +202,51 @@ void AliFemtoDreamPairCleaner::CleanDecayInvMass(std::vector<AliFemtoDreamBasePa
   if (!fMinimalBooking)
     fHists->FillDaughtersSharedDaughter(histnumber, counter);
 }
+void AliFemtoDreamPairCleaner::CleanDecayAtRandom(std::vector<AliFemtoDreamBasePart> *Decay, int histnumber)
+{
+  int counter = 0;
+  Double_t randomNumber = gRandom->Rndm();
+
+  for (std::vector<AliFemtoDreamBasePart>::iterator itDecay1 = Decay->begin();
+        itDecay1 != Decay->end(); ++itDecay1)
+  {
+    for (auto itDecay2 = itDecay1 + 1; itDecay2 != Decay->end(); ++itDecay2)
+    {
+      if (itDecay1->UseParticle() == false) 
+      {
+        break; // if particle 1 got removed in the meantime break!
+      }
+      if(itDecay2->UseParticle() == false)
+      {
+        continue; // if cleaned somewhere before, continue
```

Comment:
can you use the same structure as e.g. in CleanDecay with two continue

Reply:
also same location of if(itDecay1->UseParticle()) ? Because if particle 1 looses the selection process against particle 2 it is still evaluated against particle 3 and above and may kick them out